### DISCUSSION TITLE
About New vs. Old Drum Tracks

### DISCUSSION THREAD

**Discussion by terminator356** [muse-sequencer/muse#1135] [Team Posts]
A technical note mainly for kybos in case of confusion, but also a refresher:

**Once upon a time in MusE...**
There was only one type of drum track.
It supported one 127-note drum map for all patches.

Florian then added multi-patch drum maps.
To support, it required creating a new type of drum track called "New Drum".
Thus for several years we had **two** kinds of drum tracks - "Old" and "New".
It required the user decide which one to work with.
And some funky logic and settings to deal with it.

Then, I found a way to merge the two drum track types, giving the
  "New" drum tracks the interesting capabilities of the "Old" drum tracks,
 something that was thought not possible in the "New" drums at the time.
Now, several years later, our single drum track type **seems** to be holding up.

However, I need to warn coders that although I rigged it so that all loaded
 song files automatically converted to the unified "New" drum track type,
 and that I hid or disabled or redirected  several key "old vs. new" things,
 in the limited time I had _that is all I did_.

Thus, the entire "old vs. new" infrastructure, coding, variables etc. **are still there**.
I've just hidden it from users, showing only the unified "New" drums.

It should be removed and/or finally merged at some point.
It's a big task but a good chunk of it is more or less removing the "Old" style code,
 while other areas must be carefully merged.
Remember, some of that "Old" style code is still used, for supporting
 the old drum maps for example.

There is one area I **neglected to hide**:
There is still an area in Settings for choosing whether new drum tracks
 default to "Old" or "New".
We must hide it, but also make sure that the non-"New" settings
 are ignored if they are found in anyone's config or song files.
I believe I took steps for most of that ignoring or hiding, but 
 it is possible I neglected that area or others.

**Comment by kybos** [comment]
Yes, I noticed there was something like that, but never tried the old ones out.
Seems like the only trace is in the Global settings, those four radiobuttons on GUI behaviour tab.

**Comment by kybos** [comment]
There is also a mention on the Midi tab of Global settings, and in the Midi Export/Import settings.

**Comment by terminator356** [comment]
Oops, yeah.
[Cracks knuckles. Rolls up sleeves... then realizes he has none... ]
Got my KDevelop open, I'll take a look if no one else is.

**Comment by terminator356** [comment]
OK I have removed config importMidiNewStyleDrum and drumTrackPreference
 and related code, changed text in the midi tab, and removed settings items.
All the commented-out places (not too many) are marked with this, we can remove them later:
`// Obsolete. There is only 'New' drum tracks now.
`
Give it a try and we'll see what shakes out.

**Comment by kybos** [comment]
Nice, I think there no UI reference to the old drum tracks now.

**Comment by kybos** [comment]
And there is more room for the GUI behaviour options on the corresponding tab, which is also good...
